services/app-api/handlers/reports/approve.ts

```diff
@@ -0,0 +1,64 @@
+import handler from "../handler-lib";
+import { fetchReport } from "./fetch";
+// utils
+import dynamoDb from "../../utils/dynamo/dynamodb-lib";
+import { error, reportTables } from "../../utils/constants/constants";
+import { hasPermissions } from "../../utils/auth/authorization";
+// types
+import { StatusCodes, UserRoles } from "../../utils/types";
+
+export const approveReport = handler(async (event, context) => {
+  // Return a 403 status if the user is not an admin.
+  if (!hasPermissions(event, [UserRoles.ADMIN, UserRoles.APPROVER])) {
+    return {
+      status: StatusCodes.UNAUTHORIZED,
+      body: error.UNAUTHORIZED,
+    };
+  }
+
+  // Get current report
+  const reportEvent = { ...event, body: "" };
+  const getCurrentReport = await fetchReport(reportEvent, context);
+
+  if (!getCurrentReport.body) {
+    return {
+      status: StatusCodes.NOT_FOUND,
+      body: error.NO_MATCHING_RECORD,
+    };
+  }
+
+  // if current report exists, parse for archived status
+  if (getCurrentReport?.body) {
```

Comment:
suggestion: do we need this second `if` wrapper? `getCurrentReport.body` must exist if the first `if` skipped, right?

Reply:
@gmrabian I can possibly remove it, not sure if there was a edge case that needed this. This code was pulled from the archive.ts